src/naive/datetime/mod.rs

```diff
@@ -527,14 +519,9 @@ impl NaiveDateTime {
     /// ```
     #[must_use]
     pub const fn checked_add_signed(self, rhs: TimeDelta) -> Option<NaiveDateTime> {
-        let (time, rhs) = self.time.overflowing_add_signed(rhs);
-
-        // early checking to avoid overflow in TimeDelta::seconds
-        if rhs <= (-1 << MAX_SECS_BITS) || rhs >= (1 << MAX_SECS_BITS) {
-            return None;
-        }
-
-        let date = try_opt!(self.date.checked_add_signed(TimeDelta::seconds(rhs)));
+        let (time, rhs_days) = self.time.overflowing_add_signed(rhs);
+        let rhs_days = try_opt!(TimeDelta::try_seconds(rhs_days));
```

Comment:
It seems pretty confusing to pass something called `rhs_days` to `try_seconds()`.

Reply:
I suppose `remainder` is less confusing.

Reply:
Sounds good.